sympy/printing/repr.py

```diff
@@ -140,7 +140,13 @@ def _print_Sum2(self, expr):
                                            self._print(expr.a), self._print(expr.b))
 
     def _print_Symbol(self, expr):
-        return "%s(%s)" % (expr.__class__.__name__, self._print(expr.name))
+        d = expr._assumptions.generator
```

Comment:
I think it is ok to use the private method internal to SymPy. If someone changes it, the tests will fail.

I have no idea about `assumptions0`. I've never looked at any of this code.